studio/app/common/routers/files.py

```diff
@@ -112,3 +118,65 @@ async def create_file(workspace_id: str, filename: str, file: UploadFile = File(
         shutil.copyfileobj(file.file, f)
 
     return {"file_path": filename}
+
+
+DOWNLOAD_STATUS: Dict[str, DownloadStatus] = {}
+
+
+@router.get(
+    "/{workspace_id}/download/status",
+    response_model=DownloadStatus,
+    dependencies=[Depends(is_workspace_available)],
+)
+async def get_download_status(workspace_id: str, file_name: str):
+    filepath = join_filepath([DIRPATH.INPUT_DIR, workspace_id, file_name])
+    try:
+        return DOWNLOAD_STATUS[filepath]
+    except:  # noqa
+        raise HTTPException(status_code=404)
+
+
+@router.get(
```

Comment:
@quanpython 
Wouldn't POST be appropriate here?

Reply:
@itutu-tienday
I think POST would be more appropriate and secure. What do you think?

Reply:
@quanpython 
In terms of the meaning of the request (registering data into the system), I think POST is appropriate.
If there is little need to use GET, please use POST.